### DISCUSSION TITLE
Get asset url

### DISCUSSION THREAD

**Discussion by divdax** [statamic/cms#8707] [Q&A] [answered]
Hey! 👋

---

**// edit**
Got it. Missing url in filesystems config. 🙈

---

I've built a custom component and I want to return a collection with a teaser_image that includes the filename but not the URL. However, when I try to fetch an asset in my controller using the following PHP code:

```php
Asset::query()
    ->where('container', 'blog_images')
    ->get()->first()->url; // <- url = null
```

<img width="1056" alt="image" src="https://github.com/statamic/cms/assets/1456739/459f3977-1fad-4c98-94cf-489568da4f5a">

The URL is coming out as null. I'm wondering how I can get the asset URL. I can't use the Antlers syntax to load an image because I'm using a custom Alpine component that interacts with AJAX requests.

What am I missing?

**Comment by duncanmcclean** [comment, marked as answer]
What does your filesystems config look like (`config/filesystems.php`)? Which of those filesystems is your `blog_images` asset container using?

**Reply by jasonvarga** [reply]
(The url was missing from the filesystem config, so it was being considered a private container)

https://statamic.dev/assets#private-containers
